XamarinCommunityToolkit/Behaviors/BaseBehavior.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using Xamarin.Forms;
+using Xamarin.Forms.Internals;
+
+namespace XamarinCommunityToolkit.Behaviors
+{
+    [Preserve(AllMembers = true)]
+    public class BaseBehavior<T> : Behavior<T> where T : BindableObject
+    {
+        public T AssociatedObject { get; private set; }
+
+        protected override void OnAttachedTo(T bindable)
+        {
+            base.OnAttachedTo(bindable);
+            AssociatedObject = bindable;
+
+            if (bindable.BindingContext != null)
+            {
+                BindingContext = bindable.BindingContext;
```

Comment:
I thought that Behaviors inherits the binding context of the control they attached to